x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/transport/actions/put/TransportPutWatchAction.java

```diff
@@ -92,14 +94,20 @@ protected void doExecute(PutWatchRequest request, ActionListener<PutWatchRespons
 
                 if (isUpdate) {
                     UpdateRequest updateRequest = new UpdateRequest(Watch.INDEX, Watch.DOC_TYPE, request.getId());
-                    updateRequest.version(request.getVersion());
+                    if (request.getIfSeqNo() != SequenceNumbers.UNASSIGNED_SEQ_NO) {
```

Comment:
why is this one not guarded by 

```
if (clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.V_7_0_0)) {
```

but the one in `TransportAckWatchAction` is?

Reply:
good catch

Reply:
this one let's the user decide which feature they use. The one in the ack watch action is internal and needs to make a decision on their own.

Reply:
makes sense